### PR TITLE
Make /s/20-hour/stage/:x/puzzle/:y canonical for the K-8 script

### PR DESCRIPTION
**Pull request by joshlory**
https://github.com/code-dot-org/code-dot-org/pull/1227 added stages to the K-8 script so http://studio.code.org/s/20-hour/stage/2/puzzle/1 resolves.  However it currently redirects to the canonical URL: http://studio.code.org/s/1/level/2.

This change makes /s/20-hour/stage/:x/puzzle/:y canonical.  Once search engines update, the /s/1/level/:n links will be deprecated.

### CODE REVIEW

**Comment by laurelfan** [comment]
Yay! :)

File: dashboard/test/controllers/scripts_controller_test.rb [hunk 1]
```diff
@@ -43,10 +43,15 @@ class ScriptsControllerTest < ActionController::TestCase
   end
 
   test "should get show of k-8" do
-    get :show, id: Script::TWENTY_HOUR_ID
+    get :show, id: '20-hour'
```
Comment (laurelfan) [inline dashboard/test/controllers/scripts_controller_test.rb]: maybe have a TWENTY_HOUR_NAME constant to follow the existing pattern

**Comment by laurelfan** [comment]
looks great!

File: dashboard/test/controllers/scripts_controller_test.rb (hunk 1 above)
Comment (joshlory) [inline dashboard/test/controllers/scripts_controller_test.rb]: Good catch, this constant already exists in script.rb.  I'll update here to use it.

File: dashboard/app/helpers/levels_helper.rb [hunk 2]
```diff
@@ -10,8 +10,6 @@ def build_script_level_path(script_level)
     case script_level.script_id
     when Script::HOC_ID
       script_puzzle_path(script_level.script, script_level.chapter)
-    when Script::TWENTY_HOUR_ID
-      script_level_path(script_level.script, script_level)
     when Script::EDIT_CODE_ID
```
Comment (laurelfan) [inline dashboard/app/helpers/levels_helper.rb]: I think eventually we can also remove the special cases for hidden scripts that only we use like edit code and jigsaw

Comment (joshlory) [inline dashboard/app/helpers/levels_helper.rb]: Yep, eventually this whole `case` statement will go away.  The groundwork is in https://github.com/code-dot-org/code-dot-org/pull/1389 which removes the remaining legacy default/*.yml script definitions.

Comment (joshlory) [inline dashboard/app/helpers/levels_helper.rb]: Finally fixing in https://github.com/code-dot-org/code-dot-org/pull/1469.

**Pull request merged**